be/src/storage/txn_manager.cpp

```diff
@@ -323,6 +323,26 @@ Status TxnManager::publish_txn2(TTransactionId transaction_id, TPartitionId part
     }
 }
 
+Status TxnManager::persist_tablet_related_txns(const std::vector<TabletSharedPtr>& tablets) {
+    std::unordered_set<std::string> persited;
+    for (auto& tablet : tablets) {
+        if (tablet == nullptr) {
+            continue;
+        }
+        auto path = tablet->data_dir()->path();
+        // skip persisted meta.
+        if (persited.find(path) != persited.end()) continue;
+
+        auto st = tablet->data_dir()->get_meta()->flush();
```

Comment:
I suggest adding some metrics for trouble shooting since this a heavy operation

Reply:
`txn_persist_total` and `txn_persist_duration_us` is added to the system metric.